packages/environment-check/src/types.ts

```diff
@@ -0,0 +1,178 @@
+export interface CheckEnvironmentOptions {
+    workspaceRoots?: string[];
+    destinations?: string[];
+    credentialCallback?: (destination: Destination) => Promise<{ username: string; password: string }>;
+}
+
+export enum OutputMode {
+    Json = 'json',
+    Markdown = 'markdown',
+    Verbose = 'verbose',
+    Zip = 'zip',
+    UserDownload = 'userDownload'
+}
+
+export interface Environment {
+    developmentEnvironment: DevelopmentEnvironment;
+    platform: NodeJS.Platform;
+    versions: NodeJS.ProcessVersions;
+    basDevSpace?: string;
+}
+
+export const enum DevelopmentEnvironment {
+    BAS = 'Business Application Studio',
+    VSCode = 'Visual Studio Code'
+}
+
+export const enum Severity {
+    Info,
+    Log,
+    Warning,
+    Error
+}
+
+export interface Logger {
+    info: (text: string) => void;
+    log: (text: string) => void;
+    warning: (text: string) => void;
+    error: (text: string) => void;
+    push: (...newMessages: ResultMessage[]) => void;
+    getMessages: () => ResultMessage[];
+}
+
+export type ResultMessageText = string;
+
+export interface ResultMessage {
+    severity: Severity;
+    text: ResultMessageText;
+}
+
+export const enum UrlServiceType {
+    FullServiceUrl = 'Full Service URL',
+    CatalogServiceUrl = 'Catalog Service',
+    PartialUrl = 'Partial URL',
+    InvalidUrl = 'Invalid URL'
+}
+
+interface BasProperties {
+    usage?: string;
+    additionalData?: string;
+    sapClient?: string;
+    webIDEEnabled?: string;
+    html5DynamicDestination?: string;
+}
+
+type AuthenticationType = 'NoAuthentication' | 'BasicAuthentication' | 'OAuth2ClientCredentials';
+
+interface BasicAuthentication {
+    userName: string;
+    userPassword: string;
+}
+
+interface Credentials {
+    authentication?: AuthenticationType;
+    basicAuthentication?: BasicAuthentication;
+}
+
+export interface Destination {
```

Comment:
Is it required to redefine the types here instead of using the types from https://github.com/SAP/open-ux-tools/blob/main/packages/btp-utils/src/destination.ts

Reply:
@tobiasqueck, I think we should move to types from `@sap/bas-sdk` and not copy them to this monorepo.
But as mitigation we can switch to types defined in `@sap-ux/btp-utils` and do a central switch to `@sap/bas-sdk` from there.

Reply:
No objections (on the contrary, please, do it). My objections were against another set of types here.

Reply:
I'm confused as to what the ask is here, 
Am I to use the existing types in `@sap-ux/btp-utils` and convert the returned destinations to align with the Destination type?
Or we want to remove the destination type from this repo and use [`DestinationListInfo `](https://github.wdf.sap.corp/devx-wing/bas-sdk/blob/83dbda31d895bad359227cb35d26ce68440e5cf6/packages-sdk/bas-sdk/src/utils/destinations.ts#L85) from `@sap/bas-sdk` .
Which would require another PR in their repo to expose this type .. additionally there are properties like `sap-platform` which are not exposed by `@sap/bas-sdk` and used by functions like [`isAbapSystem`](https://github.com/SAP/open-ux-tools/blob/22836a46cd658eb119025bbeba8e5fa4c6a16dd0/packages/btp-utils/src/destination.ts#L48) . So they would also have to be requested?

Reply:
@cianmSAP for now we can use the types exposed from `@sap-ux/btp-utils` as you did already. Aligning all types with `@sap/bas-sdk` might be out of scope for this PR.

Reply:
Updated, thanks @Klaus-Keller